x2gbfs/providers/flinkster.py

```diff
@@ -0,0 +1,256 @@
+
+    FUELTYPE_TO_PROPULSION_MAPPING = {
+        'electric': 'electric',
+        'petrol': 'combustion',
+        'multifuel': 'combustion',
+        'diesel': 'combustion_diesel',
+        'hybrid': 'hybrid',
+    }
+
+    CATEGORY_PRICING_PLAN_MAPPING = {
+        '1000': 'mini',
+        '1001': 'small',
+        '1002': 'compact',
+        '1003': 'medium',
+        '1004': 'transporter',
+    }
+
+    def __init__(self, feed_config: dict):
+        self.client_id = config('FLINKSTER_CLIENT_ID')
+        self.secret = config('FLINKSTER_SECRET')
+        self.config = feed_config
+        self.pricing_plan_ids = [plan['plan_id'] for plan in feed_config['feed_data']['pricing_plans']]
+
+    def load_stations(self, default_last_reported: int) -> Tuple[Optional[Dict], Optional[Dict]]:
+        """
+        Retrieves stations from the providers API and converts them
+        into gbfs station infos and station status.
+        Returns dicts where the key is the station_id and values
+        are station_info/station_status.
+
+        For free floating only providers, this method needs not to be overwritten.
+
+        Note: station status' vehicle availabilty currently will be calculated
+        using vehicle information's station_id, in case it is defined by this
+        provider.
+        """
+
+        gbfs_station_infos_map: dict[str, dict] = {}
+        gbfs_station_status_map: dict[str, dict] = {}
+
+        for elem in self._all_areas():
+            self._extract_from_area(elem, gbfs_station_infos_map, gbfs_station_status_map, default_last_reported)
+
+        return gbfs_station_infos_map, gbfs_station_status_map
+
+    def load_vehicles(self, default_last_reported: int) -> Tuple[Optional[Dict], Optional[Dict]]:
+        """
+        Retrieves vehicles and vehicle types from provider's API and converts them
+        into gbfs vehicles, vehicle_types.
+        Returns dicts where the key is the vehicle_id/vehicle_type_id and values
+        are vehicle/vehicle_type.
+        """
+        gbfs_vehicle_types_map: dict[str, dict] = {}
+        gbfs_vehicles_map: dict[str, dict] = {}
+
+        for elem in self._all_vehicles():
+            self._extract_from_vehicle(elem, gbfs_vehicles_map, gbfs_vehicle_types_map)
+
+        return gbfs_vehicle_types_map, gbfs_vehicles_map
+
+    def _all_areas(self) -> Generator[Dict, None, None]:
+        areas = self._get_with_authorization(f'{self.api_url}/areas?size=2000&profileUid={self.profileUid}')
+        for elem in areas['_embedded']['areas']:
+            if elem['areaType'] in ['noparkingarea', 'operationarea']:
+                continue
+            yield elem
+
+    def _all_vehicles(self) -> Generator[Dict, None, None]:
+        page = 0
+        while True:
+            vehicles = self._get_with_authorization(
+                f'{self.api_url}/availableRentalObjects?expands=name,type,fillLevel,fuelType,position,category,licensePlate&size=2000&page={page}&profileUid={self.profileUid}'
```

Comment:
[minor nitpick]

Do we know if the `profileUid` will always contain only URL-safe characters?
An ID like `abc#def` would break the the fetching logic.

Reply:
The profileUid is currently fixed. Should new providers require different profileUids, this will be handled.